http/src/api.ts

```diff
@@ -96,12 +98,65 @@ export class HttpApi implements Api<ResponseStream> {
         this.sinkSubject(1, {
           ...options,
           id,
+          namespace: this.namespace,
         })
       ),
       filter(res$ => res$.request.id === id),
       flatten
     );
   }
+
+  public isolateSource(scope: Scope): HttpApi {
+    const namespace = this.namespace.concat(scope);
+
+    const source = pipe(
+      this.source,
+      filter(res$ => isPrefixOf(namespace, res$.request.namespace))
+    );
+
+    return new HttpApi(source, this.sinkSubject, this.gen, namespace);
+  }
+
+  public isolateSink(
+    sink: Producer<SinkRequest>,
+    scope: Scope
+  ): Producer<SinkRequest> {
+    return pipe(
+      sink,
+      map(req => ({
+        ...mkOpts(req.method ?? 'GET', req),
+        namespace: [scope].concat(req.namespace ?? []),
+      }))
+    );
+  }
+
+  public create(
+    source: Producer<ResponseStream>,
+    sinkSubject: Subject<SinkRequest>,
+    gen: IdGenerator
+  ): HttpApi {
+    const wrappedSubject = (t: number, d: any) => {
+      if (t === 1 && isPrefixOf(this.namespace, d.namespace)) {
+        sinkSubject(1, {
+          ...d,
+          namespace: d.namespace.slice(this.namespace.length),
+        });
+      } else sinkSubject(t as any, d);
+    };
```

Comment:
This should be a helper function (wrapping a subject). But I am not entirely sure where to put it. Either in `@cycle/run` because that is what ultimately defines the interface a driver/api has to follow, or `@cycle/utils` because it is just a helper function

Reply:
I decided that `@cycle/run` is the best place for it, as all drivers already depend on it, but I my mind `@cycle/utils` will contain stuff that you would want to use in application code, which `wrapSubject` is not